File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
       int writer=s.length()-1;
        int reader = t.length()-1;
        String temp ="";
        if(s.isEmpty()){
            return true;
        }else if(t.isEmpty()){
            return false;
        }
        while(writer>=0 && !s.isEmpty()){
            if(s.charAt(writer)==t.charAt(reader)){
                temp=t.charAt(reader) +temp;
                writer--;
            }
            if(reader==0){
                break;
            }else {
                reader--;
            }
        }
        return s.equals(temp);
    }
}

File: 0392-is-subsequence/0392-is-subsequence.cs
public class Solution {
    public bool IsSubsequence(string s, string t) {
           int first = 0 ;
           int second = 0;
            if (s.Length > t.Length) return false;
            if (s.Length == t.Length && s == t) return true; 

           while(first<s.Length && second<t.Length){
                // Console.WriteLine(first + " " + second);
               
                if(t[second] == s[first]){
                   first++;
                    second++;
                }else{
                    while(first<s.Length && second<t.Length && t[second] != s[first]) second++;
                }
           }
           
           if(first == s.Length){
                return true;
           }else{
                return false;
           }
    }
}